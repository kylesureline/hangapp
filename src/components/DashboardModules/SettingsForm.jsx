import React from 'react';
import { connect } from 'react-redux';
import { setTheme } from '../../actions/settings';
import { setGuessesRemaining } from '../../actions/player';

export class SettingsForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      error: ''
    };
  }
  onThemeChange = (e) => {
    const newTheme = e.target.value;
    this.props.setTheme(newTheme);
  };
  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <h3 className="module__title">Settings</h3>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <div>
          <label htmlFor="theme">Theme:</label><br />
  				<input
            id="light"
            type="radio"
            name="theme"
            value="light"
            checked={this.props.settings.theme === 'light'}
            onChange={this.onThemeChange} />
          <label htmlFor="light">Light</label>
  				<input
            id="dark"
            type="radio"
            name="theme"
            value="dark"
            checked={this.props.settings.theme === 'dark'}
            onChange={this.onThemeChange} />
          <label htmlFor="dark">Dark</label>
        </div>
      </form>
    );
  }
};

const mapStateToProps = (state) => ({
  settings: state.settings,
  guessesRemaining: state.player.guessesRemaining,
  guessedLetters: state.player.guessedLetters
});

const mapDispatchToProps = (dispatch) => ({
  setTheme: (theme) => dispatch(setTheme(theme))
});

export default connect(mapStateToProps, mapDispatchToProps)(SettingsForm);
