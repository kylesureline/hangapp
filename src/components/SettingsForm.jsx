import React from 'react';
import { connect } from 'react-redux';
import { setTheme, setDifficulty } from '../actions/settings';

export class SettingsForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      theme: 'light',
      difficulty: 'easy',
      error: ''
    };
  }
  onThemeChange = (e) => {
    const newTheme = e.target.value;
    this.setState({ 'theme': newTheme });
    this.props.setTheme(newTheme);
  };
  onDifficultyChange = (e) => {
    const newDifficulty = e.target.value;
    this.setState({ 'difficulty': newDifficulty });
    this.props.setDifficulty(newDifficulty);
  };
  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <h3>Settings</h3>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <div>
          <label htmlFor="theme">Theme:</label><br />
  				<input
            id="light"
            type="radio"
            name="theme"
            value="light"
            checked={this.state.theme === 'light'}
            onChange={this.onThemeChange} />
          <label htmlFor="light">Light</label>
  				<input
            id="dark"
            type="radio"
            name="theme"
            value="dark"
            checked={this.state.theme === 'dark'}
            onChange={this.onThemeChange} />
          <label htmlFor="dark">Dark</label>
        </div>
        <br /><br />
        <div>
          <label htmlFor="difficulty">Difficulty:</label><br />
          <input
            id="easy"
            type="radio"
            name="difficulty"
            value="easy"
            checked={this.state.difficulty === 'easy'}
            onChange={this.onDifficultyChange} />
          <label htmlFor="easy">Easy</label>
          <input
            id="medium"
            type="radio"
            name="difficulty"
            value="medium"
            checked={this.state.difficulty === 'medium'}
            onChange={this.onDifficultyChange} />
          <label htmlFor="medium">Medium</label>
          <input
            id="hard"
            type="radio"
            name="difficulty"
            value="hard"
            checked={this.state.difficulty === 'hard'}
            onChange={this.onDifficultyChange} />
          <label htmlFor="hard">Hard</label>
        </div>
      </form>
    );
  }
};

const mapDispatchToProps = (dispatch) => ({
  setTheme: (theme) => dispatch(setTheme(theme)),
  setDifficulty: (difficulty) => dispatch(setDifficulty(difficulty))
});

export default connect(undefined, mapDispatchToProps)(SettingsForm);
