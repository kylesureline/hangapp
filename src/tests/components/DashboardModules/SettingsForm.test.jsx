import React from 'react';
import { shallow } from 'enzyme';
import { SettingsForm } from '../../../components/DashboardModules/SettingsForm.jsx';

let setTheme, wrapper;

beforeEach(() => {
  setTheme = jest.fn();
  wrapper = shallow(
    <SettingsForm
      settings={{ theme: 'light' }}
      setTheme={setTheme}
    />
  );
});

test('should render SettingsForm correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle theme change correctly', () => {
  const e = { target: { value: 'dark' } };
  wrapper.find('#dark').prop('onChange')(e);
  expect(setTheme).toHaveBeenCalledWith(e.target.value);
});
