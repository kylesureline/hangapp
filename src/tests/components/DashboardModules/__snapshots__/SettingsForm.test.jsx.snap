// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SettingsForm correctly 1`] = `
<form
  className="form"
>
  <h3
    className="module__title"
  >
    Settings
  </h3>
  <div>
    <label
      htmlFor="theme"
    >
      Theme:
    </label>
    <input
      checked={true}
      id="light"
      name="theme"
      onChange={[Function]}
      type="radio"
      value="light"
    />
    <label
      htmlFor="light"
    >
      Light
    </label>
    <input
      checked={false}
      id="dark"
      name="theme"
      onChange={[Function]}
      type="radio"
      value="dark"
    />
    <label
      htmlFor="dark"
    >
      Dark
    </label>
  </div>
</form>
`;
